packages/jsii-spec/lib/spec.ts

```diff
@@ -508,7 +521,7 @@ export interface Method extends Documentable, Overridable, SourceLocatable {
     /**
      * The return type of the method (undefined if void or initializer)
      */
-    returns?: TypeReference;
+    returns?: TypeInstance<TypeReference>;
```

Comment:
This is the only place where `TypeInstance` makes sense to me.

Although I think `promise?: boolean` makes more sense as an attribute of `Method` to me (`async?: boolean`) than as an attribute of the return type.